synfig-studio/src/gui/app.cpp

```diff
@@ -1348,13 +1346,45 @@ App::get_default_accel_map()
 }
 
 /* === M E T H O D S ======================================================= */
+Glib::RefPtr<App>
+App::instance()
+{
+	static Glib::RefPtr<studio::App> app_reference = Glib::RefPtr<App>(new App());
+	return app_reference;
+}
+
+void
+App::run(int argc, char** argv)
+{
+	if (argc > 0) {
```

Comment:
Why not just override `local_command_line()` to get already properly prepared arguments? 

P.S. I see only `local_command_line_vfunc()` but it works too.

Reply:
local_command_line_vfunc() handles everything, even file opening and all.

Besides, AFAIK, the only parameter for synfig studio (besides filepath) is --console (MS Windows only), that we can handle with `add_main_option_entry()`.